client/src/test/java/io/pravega/client/segment/impl/SegmentOutputStreamTest.java

```diff
@@ -86,6 +87,7 @@ public class SegmentOutputStreamTest extends LeakDetectorTestSuite {
     private static final String SEGMENT = "test/0";
     private static final String TXN_SEGMENT = "scope/stream/0.#epoch.0#transaction.00000000000000000000000000000001";
     private static final int SERVICE_PORT = 12345;
+    private static final int WAIT_TILL_RECONNECT_NOT_DONE = 50;
```

Comment:
We shouldn't be doing sleep in unit tests. Let's either run with an inline executor and zeroed delayes so we can get determinism or use AssertEventuallyEquals